File: examples/spring-boot-maven-plugin/src/main/java/community/flock/wirespec/examples/app/user/UserService.java
package community.flock.wirespec.examples.app.user;

import java.util.List;

interface UserContext extends
        UserAdapter.Module {

    class UserService {
        List<User> getAllUsers(UserContext ctx) {
            return ctx.getUserAdapter().getAllUsers();
        }

        User getUserByName(UserContext ctx, String name) {
            return ctx.getUserAdapter().getUserByName(name);
        }

        User saveUser(UserContext ctx, User user) {
            return ctx.getUserAdapter().saveUser(user);
        }

        User deleteUserByName(UserContext ctx, String name) {
            return ctx.getUserAdapter().deleteUserByName(name);
        }
    }
}

File: examples/wirespec-gradle-plugin-ktor/src/main/kotlin/community/flock/wirespec/examples/app/user/UserService.kt
package community.flock.wirespec.examples.app.user

interface UserService : HasUserAdapter

fun UserService.getAllUsers() = userAdapter.getAllUsers()

fun UserService.getUserByName(name: String) = userAdapter.getUserByName(name)

fun UserService.saveUser(user: User) = userAdapter.saveUser(user)

fun UserService.deleteUserByName(name: String) = userAdapter.deleteUserByName(name)
